src/Main.hs

```diff
@@ -4,44 +4,29 @@
 module Main where
 
 import           Control.Applicative
-import           Control.Exception
 import           Control.Monad
 import           Control.Monad.Trans
 import           Data.Aeson
 import qualified Data.ByteString.Char8 as BSC
-import           Data.Maybe
-import           Data.Monoid
 import qualified Data.Text             as T
-import qualified Data.Vector           as V
 import           Nagios.Check.RabbitMQ
 import           Network.HTTP.Client
 import           System.Environment
-import           System.Exit
 import           System.Nagios.Plugin
 
+simplePerfDatum :: T.Text -> PerfValue -> NagiosPlugin()
+simplePerfDatum n p = addPerfDatum n p NullUnit Nothing Nothing Nothing Nothing
+
 main :: IO ()
 main = runNagiosPlugin $ do
-    CheckOptions{..} <- liftIO $ parseOptions
+    CheckOptions{..} <- liftIO parseOptions
 
     username <- liftIO $ maybe "" BSC.pack <$> lookupEnv "RABBIT_USER"
     password <- liftIO $ maybe "" BSC.pack <$> lookupEnv "RABBIT_PASS"
 
     manager <- liftIO $ newManager defaultManagerSettings
 
-    let baseUrl = concat [ "http://", hostname, "/api" ]
-
-    -- Get the length of the response at /api/connections (perfdata only)
-    let connUrl = concat [ baseUrl, "/connections" ]
-    connRequest <- applyBasicAuth username password <$> parseUrl connUrl
-
-    resp' <- liftIO $ httpLbs connRequest manager
-
-    let connCount = case eitherDecode (responseBody resp') of
-            Left e          -> fromIntegral 0
-	    Right (Array x) -> (fromIntegral (V.length x))
-
-    -- Full Exchange rates check
-    let rateUrl = concat [ baseUrl, "/exchanges/%2F/", exchange ]
+    let rateUrl = concat [ "http://", hostname, "/api/exchanges/%2F/", exchange ]
```

Comment:
I can remove this, but any invocations must include the scheme in the host name, otherwise there will be an exception:  `InvalidUrlException "<url>" "Invalid scheme"`. The error is pretty clear what's missing though, so that's nice.